pennychat/views.py

```diff
@@ -34,20 +55,15 @@ def get(self, request, pk, format=None):
         return Response(serializer.data)
 
     def post(self, request, pk, format=None):
+        if not request.auth:
+            raise NotAuthenticated
         follow_up_data = dict(request.data)
         penny_chat_url = reverse('pennychat-detail', args=[pk], request=request)
         follow_up_data['penny_chat'] = penny_chat_url
         serializer = FollowUpSerializer(data=follow_up_data, context={'request': request})
         if serializer.is_valid():
-            # TODO - if request.user is not anonymous then find the UserProfile corresponding to the request user
-            # creating new follow ups will not work unless you uncomment these lines!
-            # user, created = UserProfile.objects.get_or_create(
-            #     real_name='Anonymous Profile',
-            #     defaults={
-            #         'email': 'anonymous@profile.com',
-            #         'slack_team_id': '1'
-            #     })
-            # serializer.save(user=user)
+            profile = UserProfile.objects.get(user=request.user)
+            serializer.save(user_profile=profile)
```

Comment:
This will be weird at some point. When a user has more than one slack then they will have more than one UserProfile and this will actually cause an error and they will not be able to post FollowUps. Definitely don't have to solve this now, but I am puzzling about how this will change going forward. 

https://github.com/penny-university/penny_university/issues/168